idx/memory/find_cache.go

```diff
@@ -83,8 +99,11 @@ func (c *FindCache) Add(orgId uint32, pattern string, nodes []*Node) {
 			return
 		}
 		c.Lock()
-		c.cache[orgId] = cache
-		c.newSeries[orgId] = make(chan struct{}, c.invalidateQueueSize)
+		// re-check. someone else may have added a cache in the meantime.
+		_, ok := c.cache[orgId]
```

Comment:
If this is `ok`, then `cache` needs to be updated to point to the item in the map.  Otherwise the future calls to cache.Add() will add the items to the wrong lru.

Maybe, this should be
```
c.Lock()
cache, ok = c.cache[orgId]
if !ok {
  cache, err = lru.New(c.size)
  //handle err
  c.cache[orgId] = cache
}
c.Unlock()
```

Reply:
makes sense